buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java

```diff
@@ -214,6 +228,27 @@ private void traverse(
     }
   }
 
+  /**
+   * Adds all of the elements in {@code otherMap} to {@code mainMap}. If a key
+   * is already present in {@code mainMap} it is removed before being updated.
+   * This is done to ensure that the {@code filePath} gets updated if a given
+   * keyword has been overridden.
+   *
+   * @param mainMap The map that is getting entries added to it
+   * @param otherMap The map whose entries are being added from
+   */
+  private static void addOrReplaceEntries(Map<Keyword, String> mainMap,
+      final Map<Keyword, String> otherMap) {
+    for (Map.Entry<Keyword, String> entry : otherMap.entrySet()) {
+      Keyword key = entry.getKey();
+      String value = entry.getValue();
+      if (mainMap.containsKey(key)) {
+        mainMap.remove(key);
```

Comment:
Is this removal needed? My understanding is that put() overwrites any current entries with the given key.

Reply:
Unfortunately it is required. This is because put() doesn't override the key but only the value. In this case we also want to override the key since the keyword's filePath is being updated.

Reply:
Ok, SGTM